### DISCUSSION TITLE
How to serve static files on-demand after the build process

### DISCUSSION THREAD

**Discussion by calebeaires** [nitrojs/nitro#1498] [General]
In an Express.js or NestJS application, we can use the "Serving static files in Express" feature to allow a server to serve static files (such as CSV, JPG, TXT) so that the files can be added/removed after the build and during production. After studying the Nitro and Nuxt 3 documentation, I did not find a similar feature. In Nitro, both "serverAssets" and "publicAssets" are part of the build process. Here's an example of using NestJS to serve files statically on-demand, such as when a user changes their profile picture:

```ts
import { Module } from '@nestjs/common';
import { AppController } from './app.controller';
import { AppService } from './app.service';
import { ServeStaticModule } from '@nestjs/serve-static';
import { join } from 'path';

@Module({
  imports: [
    ServeStaticModule.forRoot({
      rootPath: join(__dirname, '..', 'client'),
    }),
  ],
  controllers: [AppController],
  providers: [AppService],
})
export class AppModule {}


```

**_Question:_**
Is there an alternative approach in Nitro or Nuxt 3 to serve static files on-demand after the build process?

**Comment by Hebilicious** [comment]
> In an Express.js or NestJS application, we can use the "Serving static files in Express" feature to allow a server to serve static files (such as CSV, JPG, TXT) so that the files can be added/removed after the build and during production. After studying the Nitro and Nuxt 3 documentation, I did not find a similar feature. Both "serverAssets" and "publicAssets" are part of the build process. Here's an example using NestJS to serve files statically on-demand, such as when a user changes their profile picture:
> 
> ```ts
> import { Module } from '@nestjs/common';
> import { AppController } from './app.controller';
> import { AppService } from './app.service';
> import { ServeStaticModule } from '@nestjs/serve-static';
> import { join } from 'path';
> 
> @Module({
>   imports: [
>     ServeStaticModule.forRoot({
>       rootPath: join(__dirname, '..', 'client'),
>     }),
>   ],
>   controllers: [AppController],
>   providers: [AppService],
> })
> export class AppModule {}
> ```
> 
> **_Question:_** Is there an alternative approach in Nitro or Nuxt 3 to serve static files on-demand after the build process?

Does adding your files to the `public` directory work ?

**Reply by calebeaires** [reply]
No, because many of the files are generated while using the application. For example, there could be a CSV file created to populate application resources or an image dynamically added to a folder for later access by the frontend through a direct URL like http://my-app.com/dynamic-content-assets-folder/some-file.txt.
